### PR TITLE
addresses depth and  UI  problems

### PR DESCRIPTION
**Pull request by tony1960**
This address 8811, 8809, 8898

### CODE REVIEW

File: rolocController.cpp [hunk 1]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+                float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
 
-                if(!isOutOfBounds)
+                double mean = getMean(mDepthAccumulator);
+                double stdDev = qSqrt(getVariance(mDepthAccumulator));
+#if DBG_BLOCK
+                qDebug() << "*************************Distance Stats*****************";
+                qDebug() << "*************************** bit set  -> " <<  CHECK_BIT(rolocData, 1)  << "*********************";
+                qDebug() << "smaple size: " << mNumSamples;
+                qDebug() << "sample size: " << mNumSamples;
```
Comment (dmacattack) [inline rolocController.cpp:257]: removing duplicate dbg
```suggestion
                qDebug() << "sample size: " << mNumSamples;
```

File: rolocController.cpp [hunk 2]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+            mNumSamples++;
 
-            for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
+            if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
             {
-                quint8 isLessThanLowerBound = mDepthAccumulator.at(i) < mean - stdDev * scaleOfElimination;
-                quint8 isGreaterThanUpperBound = mDepthAccumulator.at(i) > mean + stdDev * scaleOfElimination;
-                quint8 isOutOfBounds = isLessThanLowerBound || isGreaterThanUpperBound;
+                QList<quint8> acceptedDepthReadings;
+                float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
```
Comment (dmacattack) [inline rolocController.cpp:249]: c++ cast
```suggestion
                float scaleOfElimination = static_cast<float>(N_MULTI_LF_DEPTH_DELTA);
```

File: rolocController.cpp [hunk 3]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+                qDebug() << "sample size: " << mNumSamples;
+                qDebug() << "Mean: " << mean;
+                qDebug() << "Standard Dev.: " << stdDev;
+#endif
+                for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
                 {
-                    acceptedDepthReadings.append(mDepthAccumulator.at(i));
+                    quint8 isLessThanLowerBound = mDepthAccumulator.at(i) < mean - stdDev * scaleOfElimination;
+                    quint8 isGreaterThanUpperBound = mDepthAccumulator.at(i) > mean + stdDev * scaleOfElimination;
+                    quint8 isOutOfBounds = isLessThanLowerBound || isGreaterThanUpperBound;
```
Comment (dmacattack) [inline rolocController.cpp:265]: can you add parens around your math to ensure the order of ops is correct?

File: rolocController.cpp [hunk 4]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+                        acceptedDepthReadings.append(mDepthAccumulator.at(i));
+                    }
                 }
-            }
 
-            double finalReading = getMean(acceptedDepthReadings);
-            qDebug() << finalReading;
-            mROLOCdepthMeasurement = finalReading;
+                double finalReading = getMean(acceptedDepthReadings);
+                mROLOCdepthMeasurement = finalReading  * 0.0254;
```
Comment (dmacattack) [inline rolocController.cpp:274]: can you make this a constant ?

File: rolocController.cpp [hunk 5]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+                sendDataReport();
+                mNumSamples = 0;
+                mCurrentMode = ROLOC::eMODE_GET_SIGNAL_STRENGTH;
+                qDebug() << "mFrequency  mFrequency-> " << mFrequency;
```
Comment (dmacattack) [inline rolocController.cpp:278]: duplicate dbg
```suggestion
                qDebug() << " mFrequency-> " << mFrequency;
```

File: rolocController.cpp [hunk 6]
```diff
@@ -447,12 +538,41 @@ void ROLOCcontroller::onFreqSetTimerExpired()
...
+ * @brief ROLOCcontroller::rolocSetDepthMode - set depth mode
+ * @param mode - mode enum
+ */
+void ROLOCcontroller::rolocSetDepthMode(ROLOC::eLINEFINDER_MODE mode)
+{
+    Q_UNUSED(mode);
+    qint16 depthPacket = 0;
+    depthPacket = 0x0200;
+    depthPacket |= mFrequency;
+    int status = m_i2cBus.i2c_writeWord(mI2cAddr, ROLOC::eCMD_INFO, depthPacket);
```
Comment (dmacattack) [inline rolocController.cpp:551]: simplify the equation. can you make this 0x0200 a constant ?

```suggestion
    qint16 depthPacket = 0x0200 | mFrequency ;
    int status = m_i2cBus.i2c_writeWord(mI2cAddr, ROLOC::eCMD_INFO, depthPacket);
```

File: rolocController.hpp [hunk 7]
```diff
@@ -72,7 +78,10 @@ private slots:
     ROLOC::eSTATE mCurrentState;
     QTimer *mpFreqencySetTimer;
     ROLOC::eLINEFINDER_FREQ mPendingFreq;
-
+    QQueue<bool>     i2cValid;
+    int mDisplayRetry = 0;
+    int mBadReadCount = 0;
+    bool mPrevPresent = true;
```
Comment (dmacattack) [inline rolocController.hpp:84]: lets put these in the ctors initializer list

**Review by dmacattack — CHANGES_REQUESTED** [review]

File: rolocController.cpp [hunk 8]
```diff
@@ -118,37 +126,65 @@ void ROLOCcontroller::rolocBusy(ROLOC::eSTATE nextState)
...
 
-    if (mCurrentState != ROLOC::eSTATE_BUSY)
-    {
+#endif
         // update hardware existence
         bool present = rolocHardwarePresent();
-
+        if (present == false)
+         {
+           if (mBadReadCount == MAX_BAD_READS)
```
Comment (rabidmonkeyman) [inline rolocController.cpp:158]: should this be ">="?

File: rolocController.cpp [hunk 9]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+            // update the sig strength
+            mROLOCsignalStrenth = rolocData;
+#if DBG_BLOCK
+            qDebug() << "sig strength = " << mROLOCsignalStrenth; // TODO verify if is this working ?
+#endif
+            // clear the depth measurements
+            mDepthAccumulator.clear();
+            sendDataReport();
+        }
+        else
```
Comment (rabidmonkeyman) [inline rolocController.cpp:240]: is this an else if? or is it a nested if inside the else statement? the formatting is poor

File: rolocController.cpp [hunk 10]
```diff
@@ -188,50 +224,62 @@ void ROLOCcontroller::processRolocData()
...
+        if (mCurrentMode == ROLOC::eMODE_GET_DEPTH_MEASUREMENT)
+        {
+            mDepthAccumulator.append(rolocData);
+            mNumSamples++;
 
-            for(quint8 i = 0; i < mDepthAccumulator.count(); i++)
+            if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
```
Comment (rabidmonkeyman) [inline rolocController.cpp:246]: should this also be ">="?

File: rolocController.cpp [hunk 11]
```diff
@@ -381,13 +450,35 @@ double ROLOCcontroller::getVariance(QList<quint8> values)
  */
 void ROLOCcontroller::sendDataReport()
 {
-    mDbusHandler.sendDataReport(
+
+
```
Comment (rabidmonkeyman) [inline rolocController.cpp:454]: so much extra new lines

File: rolocController.cpp [hunk 12]
```diff
@@ -447,12 +538,41 @@ void ROLOCcontroller::onFreqSetTimerExpired()
...
 
+/**
+ * @brief ROLOCcontroller::rolocSetDepthMode - set depth mode
+ * @param mode - mode enum
+ */
+void ROLOCcontroller::rolocSetDepthMode(ROLOC::eLINEFINDER_MODE mode)
+{
+    Q_UNUSED(mode);
+    qint16 depthPacket = 0;
+    depthPacket = 0x0200;
```
Comment (rabidmonkeyman) [inline rolocController.cpp:549]: magic number

**Review by rabidmonkeyman — CHANGES_REQUESTED** [review]
i just want to point out that every time i make a release for i3000, this roloccontroller is always spamming the console with worthless prints. this thing is sporadically used as is, we dont need to know every time an i2c transaction with it failed because its not present, etc. please keep that in mind when working on this component. if you run this application, will it print 5 lines to the console every 5 seconds?

File: rolocController.cpp (hunk 1 above)
Comment (tony1960) [reply rolocController.cpp:257]: done

File: rolocController.cpp (hunk 3 above)
Comment (tony1960) [reply rolocController.cpp:265]: done

File: rolocController.cpp (hunk 4 above)
Comment (tony1960) [reply rolocController.cpp:274]: Yes

File: rolocController.cpp (hunk 5 above)
Comment (tony1960) [reply rolocController.cpp:278]: DOne

File: rolocController.cpp (hunk 6 above)
Comment (tony1960) [reply rolocController.cpp:551]: Yes

File: rolocController.hpp (hunk 7 above)
Comment (tony1960) [reply rolocController.hpp:84]: Done

File: rolocController.cpp (hunk 10 above)
Comment (tony1960) [reply rolocController.cpp:246]: We chatted about this

File: rolocController.cpp (hunk 11 above)
Comment (tony1960) [reply rolocController.cpp:454]: Yeah yeah fixed

File: rolocController.cpp (hunk 12 above)
Comment (tony1960) [reply rolocController.cpp:549]: Done

**Comment by dmacattack** [comment]
@tony1960 you requested us to review again, but you didnt push any new changes. Did you forget to push new changes?
Remember you can make code changes and do another git push origin <branchname>

**Comment by tony1960** [comment]
ok I got them all

File: rolocController.cpp [hunk 13]
```diff
@@ -2,10 +2,9 @@
 #include <QtMath>
 #include "rolocController.hpp"
 #include "inspRolocControllerDbus.hpp"
-#define DBG_BLOCK 0
+#define DBG_BLOCK 1
```
Comment (dmacattack) [inline rolocController.cpp:5]: Lets change this to 0 before merging

File: rolocController.cpp [hunk 14]
```diff
@@ -51,6 +52,9 @@ ROLOCcontroller::ROLOCcontroller()
 , mpRolocDataPollingTimer(NULL)
 , mCurrentState(ROLOC::eSTATE_DISCONNECTED)
 , mpFreqencySetTimer(NULL)
+, mDisplayRetry(0)
+, mBadReadCount(0)
+, mPrevPresent(true)
 , mPendingFreq(mFrequency)
 {
     qCritical() << "------------------------";
```
Comment (dmacattack) [inline rolocController.cpp:60]: Can you do me a favor ? I know this isnt your code. It was mine when the address was causing problems with linux. 

Can you delete this ? I dont think we need this anymore

File: rolocController.cpp [hunk 15]
```diff
@@ -525,7 +526,7 @@ void ROLOCcontroller::setFrequencyHandler(ROLOC::eLINEFINDER_FREQ freq)
 
 /**
  * @brief ROLOCcontroller::onFreqSetTimerExpired - timer callback when we should set the frequency to ensure
- * the ROLOC doesnt get mixed up
+ * the ROLOC doesnt get mixed uponImageEdited
```
Comment (dmacattack) [inline rolocController.cpp:528]: huh ?

File: rolocController.cpp (hunk 3 above)
Comment (tony1960) [reply rolocController.cpp:268]: as per discussion, this has not been touched

File: rolocController.cpp (hunk 2 above)
Comment (tony1960) [reply rolocController.cpp]: as per discussion, this has not been touched

File: rolocController.cpp (hunk 5 above)
Comment (tony1960) [reply rolocController.cpp]: removed

File: rolocController.cpp (hunk 8 above)
Comment (tony1960) [reply rolocController.cpp]: sure

File: rolocController.cpp (hunk 10 above)
Comment (tony1960) [reply rolocController.cpp:251]: as per discussion, this has not been touched

File: rolocController.cpp (hunk 13 above)
Comment (rabidmonkeyman) [inline rolocController.cpp:5]: you gotta leave this off before you merge

**Review by dmacattack — APPROVED** [review]
Just a few changes
* turn off the dbg_block
* looks like theres a copy paste error on line 529
* can you delete the qCritical code in the ctor

File: rolocController.cpp [hunk 16]
```diff
@@ -31,6 +31,7 @@ namespace {
     const int    TIMER_DATA_POLLING_PERIOD                  = 333;          // period between data polls in ms
     const int    TIMER_3SECONDS                             = 3000;
     const int    FREQ_SET_TIMER_INTERVAL                    = 100;
+    const int    INCHES_TO_METERS                           = 0.0254;
```
Comment (rabidmonkeyman) [inline rolocController.cpp:34]: int? this should be a float or a double

File: rolocController.cpp (hunk 10 above)
Comment (rabidmonkeyman) [reply rolocController.cpp:251]: we chatted about this?

File: rolocController.cpp [hunk 17]
```diff
@@ -237,24 +243,21 @@ void ROLOCcontroller::processRolocData()
...
+        else if (mCurrentMode == ROLOC::eMODE_GET_DEPTH_MEASUREMENT)
         {
             mDepthAccumulator.append(rolocData);
             mNumSamples++;
 
             if(mNumSamples > N_MULTI_LF_DEPTH_SAMPLE)
             {
                 QList<quint8> acceptedDepthReadings;
-                float scaleOfElimination = (float)N_MULTI_LF_DEPTH_DELTA;
+                float scaleOfElimination = static_cast<float>(N_MULTI_LF_DEPTH_DELTA);
```
Comment (rabidmonkeyman) [inline rolocController.cpp:254]: im not sure what the point of this variable is? why not just replace this variable with the constant in the equations its used in

**Review by rabidmonkeyman — CHANGES_REQUESTED** [review]

File: rolocController.cpp (hunk 16 above)
Comment (dmacattack) [reply rolocController.cpp:34]: good point. this will make this constant zero with the integer truncation.

**Review by dmacattack — APPROVED** [review]

**Review by rabidmonkeyman — APPROVED** [review]

**Pull request merged**